src/scripts/install_machine.py

```diff
@@ -0,0 +1,326 @@
+    if args['type'] == "install":
+        vnics = args['vnics'] or ['none']
+        for network in vnics:
+            params += ['--network', network]
+
+    # Consoles
+    if args['type'] == "install":
+        graphics = args['graphics'] or ['none']
+        for console in graphics:
+            params += ['--graphics', console]
+    else:
+        has_param = False
+        for graphics in get_graphics_capabilies(args['connectionName']):
+            params += ['--graphics', f"{graphics},listen=127.0.0.1"]
+            has_param = True
+
+        if not has_param:
+            params += ['--graphics', "none"]
+
+    # Installation media
+    if args['sourceType'] != "cloud":
+        prepare_installation_source(params, args)
+
+    # Autostart
+    if args.get('autostart'):
+        params.append("--autostart")
+
+    # VCPUs
+    if 'vcpu' in args:
+        params += ['--vcpus', args['vcpu']]
+
+    # Firmware
+    if 'firmware' in args:
+        params += ['--boot', args['firmware']]
+
+    prepare_unattended(params, args, files)
+    prepare_cloud_init(params, args, files)
+
+    logging.debug(params)
+
+    return params
+
+
+def cleanup(files):
+    for file in files:
+        os.unlink(file)
+
+
+def create_vm(args, files):
+    params = prepare_virt_install_params(args, files)
+
+    xml = subprocess.check_output(params)
+
+    if args['startVm']:
+        try:
+            assert_vm_exists(args['connectionName'], args['vmName'])
+
+            xml = virsh(args['connectionName'], "dumpxml", "--inactive", args['vmName'])
+        except subprocess.CalledProcessError:
+            logging.info("The VM got deleted while being installed")
+            logging.info(traceback.format_exc())
+            sys.exit(0)
+
+    xml_last_phase = xml.strip().split(b'\n\n')[-1]
+    # Get last step only - virt-install can output 1 or 2 steps
+    inject_metadata(xml_last_phase.decode())
+
+
+def install_vm(args, files):
+    # If the VM already exists and we want to run the installer we need to remove it first
+    try:
+        virsh(args['connectionName'], "destroy", args['vmName'])
+    except subprocess.CalledProcessError:
+        pass
+
+    prevXML = virsh(args['connectionName'], "dumpxml", args['vmName'])
+
+    virsh(args['connectionName'], "undefine", args['vmName'], "--managed-save")
+
+    params = prepare_virt_install_params(args, files)
+
+    try:
+        subprocess.check_output(params)
+    except subprocess.CalledProcessError as e:
+        # If virt-install returned non-zero return code, redefine
+        # the VM show that we get back the metadata which enable the 'Install'
+        # button, so that the user can re-attempt installation
+        with tempfile.NamedTemporaryFile() as file:
+            logging.debug("virt-install failed, redefining to renable the 'Install' button")
+            file.write(prevXML)
+            file.flush()
+            virsh(args['connectionName'], "define", file.name)
+            raise e
```

Comment:
Did you mean to outdent this raise? Raising the exception inside the temp file context is rather confusing.

Reply:
Yes